apps/src/studio/studio.js

```diff
@@ -787,7 +786,16 @@ var setSvgText = function (opts) {
       tspan.firstChild.data = tspan.firstChild.data.slice(0, len);
 
       if (opts.maxLines === tSpansAdded) {
-        return opts.fullHeight;
+        if (opts.width < opts.maxWidth) {
+          // Try again with a wider speech bubble
+          return setSvgText(Object.assign({}, opts, {
+            width: Math.min(opts.width * words.length / i, opts.maxWidth),
+          }));
```

Comment:
I don't think there's an easy way to modify and check the width, since modifying one line means having to rerun the word wrap logic on the succeeding lines. The recursive version seems cleaner to me, and the `while (opts.svtText.firstChild) {...}` loop at the top of the method will remove any excess `spans.